### PR TITLE
Add reverse option to switch result order

### PR DESCRIPTION
**Pull request by techjanitor**
#66

### CODE REVIEW

File: all.go [hunk 1]
```diff
@@ -134,34 +134,67 @@ func (n *Node) all(tx *bolt.Tx, info *modelInfo, ref *reflect.Value, rtyp, typ r
 
 	if bucket != nil {
 		c := bucket.Cursor()
-		for k, v := c.First(); k != nil; k, v = c.Next() {
-			if v == nil {
-				continue
+		if opts.Reverse {
+			for k, v := c.Last(); k != nil; k, v = c.Prev() {
```
Comment (bep) [inline all.go]: You need to extract a method or a func to avoid the mighty copy-paste (to keep it DRY ...)

Comment (asdine) [inline all.go]: Agreed, the copy paste won't help maintaining this part of the code

**Comment by techjanitor** [comment]
@asdine @bep so those loop control statements, continue and break, can't exist outside of the for loop in a function. to some degree there will be code duplication. what do you propose?

**Comment by bep** [comment]
It is called programming. Add a function that returns a couple of values, error, continue bool ...

**Comment by techjanitor** [comment]
ok i broke it out into two functions

File: all.go [hunk 2]
```diff
@@ -129,72 +129,43 @@ func (n *Node) All(to interface{}, options ...func(*index.Options)) error {
...
 	results := reflect.MakeSlice(reflect.Indirect(*ref).Type(), 0, 0)
 	bucket := n.GetBucket(tx, info.Name)
 
 	if bucket != nil {
 		c := bucket.Cursor()
 		if opts.Reverse {
+			// loop through the records in descending order
 			for k, v := c.Last(); k != nil; k, v = c.Prev() {
-				if v == nil {
+				c, b := loopControl(v, opts)
```
Comment (asdine) [inline all.go]: `c` shadows the cursor, and it is not clear at all what is `c` and what is `b` without looking at the implementation of `loopControl`. Also `loopControl` is a name too generic: what does `loopControl` ? can i call it outside of a loop with the right parameters? The name should be more explicit

**Comment by techjanitor** [comment]
@asdine ok i changed the names

**Comment by asdine** [comment]
Storm is now partially based on a new query system so this PR is not compatible anymore.
But the `Reverse` option has been added everywhere now, not just on the `All` method.
See ff2b8db2 and 60fc2dd.

**Pull request closed without merging**